Mathlib/Algebra/MvPolynomial/SchwartzZippel.lean

```diff
@@ -0,0 +1,146 @@
+/-
+Copyright (c) 2023 Bolton Bailey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bolton Bailey, Yaël Dillies
+-/
+import Mathlib.Algebra.MvPolynomial.Equiv
+import Mathlib.Algebra.MvPolynomial.Variables
+import Mathlib.Algebra.Polynomial.Roots
+import Mathlib.Data.Fin.Tuple.Finset
+
+/-!
+# The Schwartz-Zippel lemma
+
+This file contains a proof of the
+[Schwartz-Zippel](https://en.wikipedia.org/wiki/Schwartz%E2%80%93Zippel_lemma) lemma.
+
+This lemma tells us that the probability that a nonzero multivariable polynomial over an integral
+domain evaluates to zero at a random point is bounded by the degree of the polynomial over the size
+of the field, or more generally, that a nonzero multivariable polynomial over any integral domain
+has a low probability of being zero when evaluated at points drawn at random from some finite subset
+of the field. This lemma is useful as a probabilistic polynomial identity test.
+
+## TODO
+
+* Generalize to polynomials over arbitrary variable types
+* Prove the stronger statement that one can replace the degrees of `p` in the RHS by the degrees of
+  the maximal monomial of `p` in some lexicographic order.
+* Write a tactic to apply this lemma to a given polynomial
+* Can the RHS be strengthened to `p.support.sup fun s ↦ ∑ i, s i / #(S i)`? Namely, can the sup
+  inside the `p.degreeOf` be pulled outside?
+
+## References
+
+* [demillo_lipton_1978]
+* [schwartz_1980]
+* [zippel_1979]
+-/
+
+local notation:70 s:70 " ^^ " n:71 => Fintype.piFinset fun i : Fin n ↦ s i
+
+open Fin Finset Fintype
+
+namespace MvPolynomial
+variable {R : Type*} [CommRing R] [IsDomain R] [DecidableEq R]
+
+-- We want to be able to refer to `hp` and `hm`.
```

Comment:
I am allowed to name `hp` if it comes before the colon, so I should be allowed to name it if it comes after the colon. 

-- Yaël

Reply:
If `hp` comes before the colon, it is an argument to the theorem and is named so that you can refer to it during the proof. In this case, the argument gets named on line 56, and the variable name on line 54 is indeed unused, as the linter says. The name given here additionally doesn't seem helpful - I agree with @Command-Master's point. In particular, because of the use of the equation compiler here, coming before or after the colon is not a fair comparison.

You could alternatively write `∀ {n} {p : MvPolynomial (Fin n) R} (S : Fin n → Finset R), p ≠ 0 →` and avoid the underscore altogether.

Can you amend the comment on line 46 to give a justification for why disabling the linter here is necessary?

Reply:
Yes but I am takin the pov of the api here. A user should be able to provide `hp` as a named argumznt regardless of whether i have used pattern-maching or `induction` to prove my lemma

Reply:
In that case, please amend the comment on line 46 with this.